### ISSUE TITLE
[RC5 IVY] Nested SVG doesn't work anymore

### ISSUE THREAD

**Issue by Yohandah** [angular/angular#34256]
# 🐞 bug report

### Affected Package
I think it's in IVY .. 

### Is this a regression?

YES.
It is working when you disable IVY with Angular 9, and it's working with Angular 8 (without Ivy)

### Description

I have a project that generate a SVG to reflect the state of an hydraulic powerplant. There is a root <SVG> tag then Angular dynamically adds component into it, that are SVG as well.

The structure of the HTML is something like 
```
<body>
<my-app>
  <svg>
    <g>
      <svg>
        <svg>
          <rect></rect>
        </svg>
      </svg>
    </g>
  </svg>
</my-app>
</body>
```

With IVY enabled, you can see the svg in the inspector but it is not rendered by the browser. If you disable IVY it works.

## 🔬 Minimal Reproduction

Since IVY is not supported by Stackblitz, here's a repo : https://github.com/Yohandah/Nested-SVG-Angular-9-IVY 

Steps :

- Clone the repo
- Install dependencies (npm install)
- run `ng serve`
- You can see a blank page, but if you inspect, you will see the SVG
- Cancel `ng serve`
- add the following to `tsconfig.json`

```
"compilerOptions": {.....},
"angularCompilerOptions": {
  "enableIvy": false
}
```
- run `ng serve` 
- You can see the SVG produced (with your eyes, and in the inspector)

## 🔥 Exception or Error
there is no error

## 🌍  Your Environment

**Angular Version:** 
<pre><code>
Angular CLI: 9.0.0-rc.5
Node: 12.13.1
OS: win32 x64

Angular: 9.0.0-rc.5
... cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.5
@angular-devkit/build-angular     0.900.0-rc.5
@angular-devkit/build-optimizer   0.900.0-rc.5
@angular-devkit/build-webpack     0.900.0-rc.5
@angular-devkit/core              9.0.0-rc.5
@angular-devkit/schematics        9.0.0-rc.5
@ngtools/webpack                  9.0.0-rc.5
@schematics/angular               9.0.0-rc.5
@schematics/update                0.900.0-rc.5
rxjs                              6.5.3
typescript                        3.6.4
webpack                           4.41.2

</code></pre>

**Comment by kara** [comment]
Looks like we aren't adding the `base-rectangle` attribute in the Ivy case, which is what's breaking the rendering. We should be adding attributes based on the selector when components are created dynamically

**Comment by AndrewKushnir** [comment]
Hi @Yohandah, thanks for reporting the issue.

I've created a [PR with proposed fix](https://github.com/angular/angular/pull/34481). PR still requires review and the final implementation might change, but I'd like to ask if you could test your app with the changes from that PR to make sure the fix covers the case that you have in the app and everything works as expected. You can follow an instruction in [this document on how to use PR artifacts](https://github.com/angular/angular/blob/master/docs/DEVELOPER.md#archives-for-each-package). An archive with updated `@angular/core` package is available [here](https://609640-24195339-gh.circle-artifacts.com/0/angular/core-pr34481-4362bd1a7f.tgz) (please make sure you update other `@angular` packages to `9.0.0-rc.13` to avoid version conflicts).

Thank you.

**Comment by Yohandah** [comment]
Hi @AndrewKushnir !

Thanks for the fix, trying this ASAP and getting back to you

**Comment by Yohandah** [comment]
@AndrewKushnir 

So I've tested with the repo that I gave you in my original post and it doesn't fix it ... 

I've updated the package.json with the following : 

```
{
  "name": "angular",
  "version": "0.0.0",
  "private": true,
  "dependencies": {
    "@angular/common": "^9.0.0-rc.13",
    "@angular/compiler": "^9.0.0-rc.13",
    "@angular/core": "C:/Users/yohan/Downloads/core-pr34481-4362bd1a7f.tgz",
    "@angular/forms": "^9.0.0-rc.13",
    "@angular/platform-browser": "^9.0.0-rc.13",
    "@angular/platform-browser-dynamic": "^9.0.0-rc.13",
    "@angular/router": "^9.0.0-rc.13",
    "core-js": "2",
    "rxjs": "^6.5.2",
    "tslib": "^1.10.0",
    "zone.js": "^0.9.1"
  },
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.900.0-rc.12",
    "@angular/cli": "~9.0.0-rc.12",
    "@angular/compiler-cli": "^9.0.0-rc.13",
    "@angular/language-service": "^9.0.0-rc.13",
    "@types/node": "~12.11.1",
    "@types/jasmine": "~2.8.8",
    "@types/jasminewd2": "~2.0.3",
    "codelyzer": "~4.5.0",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~3.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~1.1.2",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.11.0",
    "typescript": "~3.7.5"
  }
}

```

**Comment by AndrewKushnir** [comment]
Hi @Yohandah, thanks for testing your use-case with the changes from the mentioned PR. I tried to reproduce the problem and I can see the "base-rectangle" attribute being added now (with the changes from the PR), but <svg> element still doesn't show up. I performed further investigation and found the origin of the problem: the namespace is ignored when we try to create a host element for a dynamic component. I've created [a new PR](https://github.com/angular/angular/pull/35136) to add namespaces as needed. Could you please try to reproduce the problem again with the version of `@angular/core` package from that PR that [can be found here](https://612284-24195339-gh.circle-artifacts.com/0/angular/core-pr35136-8419aba29d.tgz) and let me know if the problem is resolved? Thank you.

**Comment by Yohandah** [comment]
@AndrewKushnir The problem is resolved ! Thank you ! Looking forward to the final release of Angular 9

**Comment by AndrewKushnir** [comment]
Hi @Yohandah, thanks for checking the behavior and confirming that the problem is resolved. [The PR](https://github.com/angular/angular/pull/35136) will go through the regular review and testing procedures and will be included into one of the next releases. I will keep this thread updated. Thank you.

**Comment by AndrewKushnir** [comment]
Hi, the changes from the mention PR were included into `9.0.1` release of Angular. Feel free to reopen the ticket if the problem still exists. Thank you.

**Comment by heja1600** [comment]
Add @Directive() to inherited components that has @Input() or @Output()
